unstructured/__init__.py

```diff
@@ -0,0 +1,5 @@
+from unstructured.logger import initialize_logger
+from unstructured.utils import initialize_rich_tracebacks
+
+initialize_logger(__name__)
+initialize_rich_tracebacks()
```

Comment:
So, the library itself should not be doing any logging initialization as that is in the domain of end-applications which are going to have different opinions on how to do that.

ref: https://docs.python.org/3/howto/logging-cookbook.html#adding-handlers-other-than-nullhandler-to-a-logger-in-a-library

Everywhere else in the library should just import the `logger` from `unstructured.logger` and use that to log message.

https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/main.py when invoked as a command is fair game i think, though, because it is an actual application. Though, dumb terminals should still work OK.

Reply:
> So, the library itself should not be doing any logging initialization as that is in the domain of end-applications which are going to have different opinions on how to do that.

> ref: https://docs.python.org/3/howto/logging-cookbook.html#adding-handlers-other-than-nullhandler-to-a-logger-in-a-library

Understood, I'll revert those changes.

> Everywhere else in the library should just import the logger from unstructured.logger and use that to log message.

I have always understood `logger = logging.getLogger(__name__)` to be the correct approach for logging, e.g. as suggested on the [Advanced Logging Tutorial](https://docs.python.org/3/howto/logging.html#logging-advanced-tutorial). The logger at e.g. `"unstructured.partition.auto"` will propagate the logs to the core logger at `"unstructured"`.
However, the docs on [logging for a library](https://docs.python.org/3/howto/logging.html#library-config) are not very clear on this matter. 

I can revert to using a single logger from `unstructured.logger` and import that throughout the project.

Reply:
Should be all set now. I appreciate the valuable review! It's always good to learn something new :)

Reply:
For sure, I appreciate PR's intending to better the code base!